keychain/src/keychain.rs

```diff
@@ -42,29 +43,36 @@ impl From<extkey::Error> for Error {
 	}
 }
 
-
 #[derive(Clone, Debug)]
 pub struct Keychain {
 	secp: Secp256k1,
 	extkey: extkey::ExtendedKey,
-
-	/// for tests and burn only, associate the zero fingerprint to a known
-	/// dummy private key
-	pub enable_burn_key: bool,
+	key_overrides: HashMap<Identifier, SecretKey>,
 }
 
 impl Keychain {
 	pub fn root_key_id(&self) -> Identifier {
 		self.extkey.root_key_id.clone()
 	}
 
+	// for tests and burn only, associate the zero fingerprint to a known
```

Comment:
How about making it more flexible by allowing to provide the Identifier that will go in the map as well? May be useful in unit tests, to have multiple outputs to different parties for example.